modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/semanticui/elements/icon/Icon.scala

```diff
@@ -0,0 +1,73 @@
+package edu.gemini.seqexec.web.client.semanticui.elements.icon
+
+import japgolly.scalajs.react.vdom.prefix_<^._
+import japgolly.scalajs.react.{ReactComponentB, ReactNode}
+
+/**
+  * Semantic UI Icon component
+  */
+object Icon {
+  sealed trait Flipped
+  case object NotFlipped extends Flipped
+  case object Horizontally extends Flipped
+  case object Vertically extends Flipped
+
+  sealed trait Rotated
+  case object NotRotated extends Rotated
+  case object Clockwise extends Rotated
+  case object CounterClockwise extends Rotated
+
+  case class Props(id: String,
+                   disabled: Boolean = false,
+                   loading: Boolean = false,
+                   fitted: Boolean = false,
+                   tiny: Boolean = false,
+                   mini: Boolean = false,
+                   small: Boolean = false,
+                   large: Boolean = false,
+                   big: Boolean = false,
+                   huge: Boolean = false,
+                   massive: Boolean = false,
+                   link: Boolean = false,
+                   flipped: Flipped = NotFlipped,
+                   rotated: Rotated = NotRotated,
+                   circular: Boolean = false,
+                   bordered: Boolean = false,
+                   inverted: Boolean = false,
+                   color: Option[String] = None)
```

Comment:
I guess for the size we could make it into objects. In general they are booleans that decide the presence of a given css class. In principle you could make an Icon both `tiny massive` though it probably make sense to forbid it. All the other attributes seem to be independent